src/main/java/co/cask/hydrator/plugin/batch/DynamoDBBatchSource.java

```diff
@@ -0,0 +1,220 @@
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.cdap.api.dataset.lib.KeyValue;
+import co.cask.cdap.etl.api.Emitter;
+import co.cask.cdap.etl.api.PipelineConfigurer;
+import co.cask.cdap.etl.api.batch.BatchSource;
+import co.cask.cdap.etl.api.batch.BatchSourceContext;
+import co.cask.hydrator.common.ReferencePluginConfig;
+import com.amazonaws.services.dynamodbv2.document.Item;
+import com.google.common.base.Strings;
+import org.apache.hadoop.io.LongWritable;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.annotation.Nullable;
+
+/**
+ * Batch DynamoDB Source Plugin - Reads the data from AWS DynamoDB table.
+ */
+@Plugin(type = "batchsource")
+@Name("DynamoDB")
+@Description("DynamoDB Batch Source that will read the data items from AWS DynamoDB table and emit each item as a " +
+  "JSON string, in the body field of the StructuredRecord, that can be further processed downstream in the pipeline.")
+public class DynamoDBBatchSource extends BatchSource<LongWritable, Item, StructuredRecord> {
+  private static final Schema OUTPUT_SCHEMA = Schema.recordOf(
+    "output", Schema.Field.of("body", Schema.of(Schema.Type.STRING)));
+  private final DynamoDBConfig config;
+
+  public DynamoDBBatchSource(DynamoDBConfig config) {
+    this.config = config;
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    config.validateTableName();
+    pipelineConfigurer.getStageConfigurer().setOutputSchema(OUTPUT_SCHEMA);
+  }
+
+  @Override
+  public void prepareRun(BatchSourceContext context) throws Exception {
+    context.setInput(Input.of(config.referenceName, new DynamoDBInputFormatProvider(config)));
+  }
+
+  @Override
+  public void transform(KeyValue<LongWritable, Item> input, Emitter<StructuredRecord> emitter) throws Exception {
+    StructuredRecord.Builder builder = StructuredRecord.builder(OUTPUT_SCHEMA);
+    builder.set("body", input.getValue().toJSONPretty());
+    emitter.emit(builder.build());
+  }
+
+  /**
+   * Config class for DynamoDBBatchSource.
+   */
+  public static class DynamoDBConfig extends ReferencePluginConfig {
+
+    @Description("Access key for AWS DynamoDB to connect to. (Macro Enabled)")
+    @Macro
+    private String accessKey;
+    @Description("Secret access key for AWS DynamoDB to connect to. (Macro Enabled)")
+    @Macro
+    private String secretAccessKey;
+    @Description("The region for AWS DynamoDB to connect to. Default is us_west_2 i.e. US West (Oregon).")
+    @Nullable
+    private String regionId;
+    @Description("The hostname and port for AWS DynamoDB instance to connect to, separated by a colon. For example, " +
+      "localhost:8000.")
+    @Nullable
+    private String endpointUrl;
+    @Description("The DynamoDB table to read the data from.")
+    private String tableName;
```

Comment:
Enabling macros for tableName, endpointUrl and regionID, might change the other query parameters and schema and may result into the new pipeline which user could clone and create. Hence not kept these parameters as macro enabled.

Please let me know your thoughts. Should we make these parameters macro enabled also?